edalize/gatemate.py

```diff
@@ -103,28 +103,32 @@ def configure_main(self):
         commands = EdaCommands()
         commands.commands = yosys.commands
 
-        # PnR & image generation
-        commands.add_var("P_R := $(shell which p_r)")
-        targets = self.name + "_00.cfg.bit"
+        # nextpnr & image generation
+        commands.add_var("NEXTPNR := $(shell which nextpnr-himbaechel)")
+        cfg_target = self.name + ".cfg"
         command = [
-            "$(P_R)",
-            "-A",
-            device_number,
-            "-i",
-            synth_out,
-            "-o",
-            self.name,
-            "-lib",
-            "ccag",
-            " ".join(self.tool_options.get("p_r_options", "")),
+            "$(NEXTPNR)",
+            "--device",
+            device,
+            "--json",
+            "$<",
+            "-o out=" + cfg_target,
+            " ".join(self.tool_options.get("nextpnr_options", "")),
         ]
         if ccf_file is not None:
-            command += ["-ccf", ccf_file]
+            command += ["-o ccf=" + ccf_file]
 
-        if p_r_log_file is not None:
-            command += [">", p_r_log_file]
+        if nextpnr_log_file is not None:
+            command += ["--log", nextpnr_log_file]
 
-        commands.add(command, [targets], [synth_out])
+        commands.add(command, [cfg_target], [synth_out])
 
-        commands.set_default_target(targets)
+        commands.add_var("GMPACK := $(shell which gmpack)")
```

Comment:
Use gmpack directly. We can use an EDALIZE_LAUNCHER to override the name if we want instead.